### PR TITLE
Implement `cursor::count_between`

### PR DESCRIPTION
**Pull request by agathazeren**
Count the boundaries between two indicies.

### CODE REVIEW

File: rust/element/src/cursor.rs [hunk 1]
```diff
@@ -157,6 +157,22 @@ impl<'a> Cursor<'a> {
         }
     }
 
+
+    /// Count the number of boundaries between two points.
```
Comment (appetrosyan) [inline rust/element/src/cursor.rs:161]:
```suggestion
    /// Count the number of boundaries between two points. See [`Cursor::is_boundary`].
```

File: rust/element/src/cursor.rs [hunk 2]
```diff
@@ -519,6 +535,13 @@ mod test {
         assert!(!cursor.is_boundary::<BaseMetric>());
     }
 
+    #[test]
+    fn count_between() {
+        let input = "\n \n \n";
+        let cursor = Cursor::new(&input, 0);
+        assert_eq!(3,cursor.count_between::<LinesMetric>(0, 5));
+    }
```
Comment (appetrosyan) [inline rust/element/src/cursor.rs:543]: I'd expand this test. Typically you want to include edge cases, such as empty strings, negative cases, _etc._

**Review by appetrosyan — APPROVED** [review]
LGTM
